src/slider.jsx

```diff
@@ -323,17 +314,84 @@ const Slider = React.createClass({
       document.addEventListener('touchcancel', this._dragTouchEndHandler, false);
     }
     this._onDragStart(e);
+
+    // Cancel scroll and context menu
+    e.preventDefault();
   },
 
   _onHandleMouseDown(e) {
     if (document) {
       document.addEventListener('mousemove', this._dragHandler, false);
       document.addEventListener('mouseup', this._dragEndHandler, false);
-      this._toggleSelection('none');
+
+      // Cancel text selection
+      e.preventDefault();
+
+      // Set focus manually since we called preventDefault()
+      ReactDOM.findDOMNode(this.refs.handle).focus();
```

Comment:
To be honest I never understood why the callback approach would be better, but I think I did now:

> Note that when the referenced component is unmounted and whenever the ref changes, the old ref will be called with null as an argument. This prevents memory leaks in the case that the instance is stored,...

They are trying to be sure we will not leak memory